server/data/tokenStore.ts

```diff
@@ -1,21 +1,18 @@
-import { promisify } from 'util'
-import { createRedisClient, RedisClient } from './redisClient'
+import { createRedisClient } from './redisClient'
 
 export default class TokenStore {
-  private getRedisAsync: (key: string) => Promise<string>
+  private client
 
-  private setRedisAsync: (key: string, value: string, mode: string, durationSeconds: number) => Promise<void>
-
-  constructor(redisClient: RedisClient = createRedisClient()) {
-    this.getRedisAsync = promisify(redisClient.get).bind(redisClient)
-    this.setRedisAsync = promisify(redisClient.set).bind(redisClient)
+  constructor(redisClient = createRedisClient()) {
+    this.client = redisClient
   }
 
   public async setToken(key: string, token: string, durationSeconds: number): Promise<void> {
-    this.setRedisAsync(key, token, 'EX', durationSeconds)
+    // For redis v4, the expiration is set via an options object:
+    await (await this.client).set(key, token, { EX: durationSeconds })
   }
 
-  public async getToken(key: string): Promise<string> {
-    return this.getRedisAsync(key)
+  public async getToken(key: string): Promise<string | null> {
+    return (await this.client).get(key)
```

Comment:
think there shouldn't be an await on the client its self

`await this.client.get(key)`
`await this.client.set(key, token, { EX: durationSeconds })